MusicNotationKit/MusicNotationKit/Key.swift

```diff
@@ -18,3 +18,11 @@ public struct Key {
 		self.type = type
 	}
 }
+
+extension Key: CustomDebugStringConvertible {
+	public var debugDescription: String {
+		let result:String = String(format:"key(%@ %@)",
+			String(self.noteLetter), String(self.type))
+		return result
+	}
```

Comment:
I will give a few separate comments and then give you an updated version.

- You don't need to specify the type like `result: String`. Swift infers the type. Instead just use let result = ...
- Using `self.` as a prefix is not necessary unless to disambiguate or within a closure. Therefore, it you should prefer no prefix usually.
- When creating a string like this, you can just use string interpolation like so:
 `"key(\(noteLetter) \(type))"`
- I don't think we need to make a local variable. Let's just return.

With all of these changes, here is what I propose:

```swift
public var debugDescription: String {
    return "key (\(noteLetter) \(type))"
}
```